### ISSUE TITLE
dataID.substring called on potentially numerical values

### ISSUE THREAD

**Issue by Aweary** [facebook/relay#376]
```js
  isClientID: function(dataID: string): boolean {
    return dataID.substring(0, 7) === 'client:';
  },
```

I had an instance where `dataID` was a being returned as a number from the server and this caused Relay to throw an error.

It should either check that it's a string or coerce. 

```js
 return typeof dataID === 'string' && dataID.substring(0, 7) === 'client:';
/* or */
 return (dataId + '').substring(0, 7) === 'client:';
```

Or if a numerical ID is something Relay doesn't want to support it should throw a more descriptive error.
I had a pull request ready using the `typeof` check, but figured I'd check with the contributors first.

**Comment by josephsavona** [comment]
Yes, we should have a better error message to make it clear that `id` values are required to be strings (per the [spec](http://facebook.github.io/relay/docs/graphql-object-identification.html#content)). The place to check would be in the response processing phase - `RelayQueryWriter`, `RelayNodeInterface`, maybe `forEachRootCallArg` (anywhere we grab `result.id` from the raw payload) - and add an `invariant` that `id` is a string.

**Comment by josephsavona** [comment]
@Aweary thanks for filing this, good catch!

**Comment by Aweary** [comment]
Awesome, I'll get on this and get a PR going :+1:

**Comment by josephsavona** [comment]
This was addressed in c54d49d62ac05095fe3fea8325aa2d92c7ca41e8
